add-on/src/lib/ipfs-request.js

```diff
@@ -144,7 +144,11 @@ export function createRequestModifier (getState, dnslinkResolver, ipfsPathValida
 
       // When local IPFS node is unreachable , show recovery page where user can redirect
       // to public gateway.
-      if (!state.nodeActive && request.type === 'main_frame' && sameGateway(request.url, state.gwURL)) {
+      if (!state.nodeActive &&
+        !state.redirect &&
+        request.type === 'main_frame' &&
+        sameGateway(request.url, state.gwURL)
+      ) {
```

Comment:
Please add a comment for each of the cases tested in that statement, so future generations can understand what's going on and why.